ampd/src/event_sub/mod.rs

```diff
@@ -53,109 +72,38 @@ impl<T: TmClient + Sync> EventPublisher<T> {
         (publisher, subscriber)
     }
 
-    pub async fn run(mut self, token: CancellationToken) -> Result<(), EventSubError> {
-        let mut curr_block_height = self.latest_block_height().await?;
-        let mut interval = time::interval(self.poll_interval);
+    pub async fn run(self, token: CancellationToken) -> Result<(), Error> {
+        let mut event_stream =
+            event_stream::event_stream(&self.tm_client, self.poll_interval, 10000, || {
+                self.tx.receiver_count() == 0
+            })
+            .await?;
 
         loop {
             select! {
-                _ = interval.tick() => {
-                    curr_block_height = self.process_blocks_from(curr_block_height, &token).await?.increment();
+                event = event_stream.next() => match event {
+                    Some(event) => {
+                        let event = event
+                            .inspect_err(|err| {
+                                error!(err = LoggableError::from(err).as_value(), "failed to subscribe to events");
```

Comment:
this is less a subscription error and more of a "couldn't get specific event" error, isn't it?